src/mono/mono/mini/aot-compiler.c

```diff
@@ -13098,7 +13097,7 @@ collect_methods (MonoAotCompile *acfg)
 			}
 		}
 
-		if ((method->is_generic || mono_class_is_gtd (method->klass)) && should_emit_gsharedvt_method (acfg, method)) {
```

Comment:
I think this now needs an `else if (!is_ok (error))` branch.  at the very least to call `mono_error_cleanup`, but also maybe to show a warning